extensions/interactions/CodeRepl/directives/CodeRepl.js

```diff
@@ -34,216 +34,231 @@ oppia.directive('oppiaInteractiveCodeRepl', [
       templateUrl: UrlInterpolationService.getExtensionResourceUrl(
         '/interactions/CodeRepl/directives/' +
         'code_repl_interaction_directive.html'),
-      controller: ['$scope', '$attrs', function($scope, $attrs) {
-        $scope.interactionIsActive = ($scope.getLastAnswer() === null);
-
-        $scope.$on(EVENT_NEW_CARD_AVAILABLE, function(evt, data) {
-          $scope.interactionIsActive = false;
-        });
-        $scope.language = HtmlEscaperService.escapedJsonToObj(
-          $attrs.languageWithValue);
-        $scope.placeholder = HtmlEscaperService.escapedJsonToObj(
-          $attrs.placeholderWithValue);
-        $scope.preCode = HtmlEscaperService.escapedJsonToObj(
-          $attrs.preCodeWithValue);
-        $scope.postCode = HtmlEscaperService.escapedJsonToObj(
-          $attrs.postCodeWithValue);
-
-        // Make sure $scope.preCode ends with a newline:
-        if ($scope.preCode.trim().length === 0) {
-          $scope.preCode = '';
-        } else if (!$scope.preCode.slice(-1) === '\n') {
-          $scope.preCode += '\n';
-        }
-
-        // Make sure $scope.placeholder ends with a newline.
-        if (!$scope.placeholder.slice(-1) === '\n') {
-          $scope.placeholder += '\n';
-        }
+      controller: [
+        '$scope', '$attrs', 'WindowDimensionsService',
+        'UrlService', 'EVENT_PROGRESS_NAV_SUBMITTED',
+        function(
+          $scope, $attrs, WindowDimensionsService,
+          UrlService, EVENT_PROGRESS_NAV_SUBMITTED) {
+          $scope.isSubmitHidden = function() {
+            return (!UrlService.isIframed() &&
```

Comment:
Here and below: if a bracketed phrase spans multiple lines, break after the first '(':

```
            return (
              !UrlService.isIframed() &&
              WindowDimensionsService.isWindowNarrow());
```